universal-application-tool-0.0.1/app/views/dev/AwsStorageDevViewStrategy.java

```diff
@@ -0,0 +1,60 @@
+package views.dev;
+
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.text;
+import static j2html.attributes.Attr.ENCTYPE;
+
+import j2html.TagCreator;
+import j2html.tags.ContainerTag;
+import services.cloud.StorageUploadRequest;
+import services.cloud.aws.SignedS3UploadRequest;
+import views.HtmlBundle;
+import views.ViewUtils;
+
+/** Strategy class for creating a file upload form for AWS. */
+public class AwsStorageDevViewStrategy implements CloudStorageDevViewStrategy {
+
+  @Override
+  public ContainerTag getFileUploadForm(
+      ViewUtils viewUtils, StorageUploadRequest storageUploadRequest, HtmlBundle bundle)
+      throws RuntimeException {
+    if (!(storageUploadRequest instanceof SignedS3UploadRequest)) {
+      throw new RuntimeException(
+          "Trying to upload a file to AWS dev file storage using incorrect upload request type.");
```

Comment:
suggestion (blocking): include "localstack" in the error message to clarify e.g. "...file to localstack (AWS emulator)..."